tests/ssh_tests.py

```diff
@@ -83,13 +85,23 @@ def trying_to_use_ssh_shell_after_exit_results_in_error():
     assert_raises(Exception, lambda: shell.run(["true"]))
 
 
-def _create_shell_with_wrong_port():
+@istest
+def passing_a_custom_socket_like_a_pre_connected_one():
+    sock = socket.socket(socket.AddressFamily.AF_INET, socket.SOCK_STREAM)
+    retry_on_signal(lambda: sock.connect((HOSTNAME, PORT)))
+
+    with _create_shell_with_wrong_port(sock=sock) as shell:
+        shell.run(["true"])
```

Comment:
It might be worth trying a command that produces output, such as `echo`, (and asserting that output) just to make sure output is also working correctly, similarly to one of the tests common both to SSH and local shells.